core/src/query.rs

```diff
@@ -13,47 +17,92 @@ pub struct MongoQuery {
 }
 
 impl MongoQuery {
-    // Create a new MongoStatement with StmtKind::Query on the connection currentDB.
-    // Executes a $sql aggregation with the given query and initialize the Resultset cursor.
-    // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
-    // timeout, the query must finish before the timeout or an error is returned
+    // Create a new MongoQuery on the connection's current database. Execute a
+    // $sql aggregation with the given query and initialize the result set
+    // cursor. If there is a timeout, the query must finish before the timeout
+    // or an error is returned.
     pub fn execute(
-        _client: &MongoConnection,
-        _query_timeout: Option<i32>,
-        _query: &str,
+        client: &MongoConnection,
+        query_timeout: Option<i32>,
+        query: &str,
     ) -> Result<Self> {
-        unimplemented!()
+        match &client.current_db {
+            None => Err(Error::NoDatabase),
+            Some(current_db) => {
+                let db = client.client.database(current_db);
+
+                // 1. Run the $sql aggregation to get the result set cursor.
+                let pipeline = vec![doc! {"$sql": {
+                    "dialect": "mongosql",
+                    "format": "odbc",
+                    "statement": query,
+                }}];
+
+                let options = query_timeout.map(|i| {
+                    AggregateOptions::builder()
+                        .max_time(Duration::from_millis(i as u64)) // TODO: should timeout come from client if present?
```

Comment:
The function takes a `query_timeout` argument, but the `client` also contains a timeout. Which one should we use here?

Reply:
`query_timeout` seems correct to me since that's explicitly set.  My understanding is that the query_timeout here would come from setting the `SQL_ATTR_QUERY_TIMEOUT` attribute whereas the client timeout would be from `SQL_ATTR_CONNECTION_TIMEOUT` which is not what we want here.